packages/server/src/service/firestore.ts

```diff
@@ -122,6 +122,24 @@ export class FirestoreService {
     await this.db.collection('StateInfo').doc(id).update(update)
   }
 
+  async batchUpdateRegistrations(
+    data: Array<Partial<RichStateInfo> & { id: string }>
+  ): Promise<void> {
+    // batch can only take up to 500 operations, slicing data array accordingly
+    // https://firebase.google.com/docs/firestore/manage-data/transactions#batched-writes
```

Comment:
Cosider using https://www.npmjs.com/package/array-chunk or `_.chunk` from https://underscorejs.org/

Because this is a server -- we don't really worry about importing libraries